### PR TITLE
Scanbuild null fixes

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes all `NULL` dereference issues found by clang scanbuild
* Fixes all uninitialized value issues found by clang scanbuild

### CODE REVIEW

File: client/X11/xf_event.c [hunk 1]
```diff
@@ -477,6 +477,9 @@ static BOOL xf_event_KeyRelease(xfContext* xfc, XEvent* event, BOOL app)
 }
 static BOOL xf_event_FocusIn(xfContext* xfc, XEvent* event, BOOL app)
 {
+	if (!xfc || !event || !xfc->window)
+		return FALSE;
```
Comment (hardening) [inline client/X11/xf_event.c]: how the hell can we have `xfc == NULL` or `event == NULL`, if this happens I guess this is a problem with the caller (static function). Even the check on `xfc->window` is questionable.

**Comment by hardening** [comment]
As usual I'm not sure it's a good idea to make static analyzers happy at all costs when they report false positives

**Comment by akallabeth** [comment]
@hardening These issues all have proper stack traces by the analyzer, they are correct.
The 'false'  positives are the 4 memory leaks that result from the invalid HANDLE values that are not NULL

**Comment by akallabeth** [comment]
@hardening removed the checks for `xfc` and `event` as requested.
The rest of the changes are quite valid (uninitialized memory, ...)

**Pull request merged**